jsapp/js/components/languages/languageSelector.tsx

```diff
@@ -176,27 +176,25 @@ class LanguageSelector extends React.Component<
     }
   }
 
-  fetchSuggestedLanguages() {
-    this.setState({suggestedLanguages: undefined});
+  async fetchSuggestedLanguages() {
+    this.setState({ suggestedLanguages: undefined });
     if (this.props.suggestedLanguages) {
-      this.props.suggestedLanguages.forEach(async (languageCode) => {
-        try {
-          const language = await languagesStore.getLanguage(languageCode);
-          // Just a safe check if suggested languages list didn't change while we
-          // waited for the response.
-          const isAlreadyAdded = Boolean(this.state.suggestedLanguages?.find((stateLanguage) => stateLanguage.code === language.code));
-          if (
-            this.props.suggestedLanguages?.includes(language.code) &&
-            !isAlreadyAdded
-          ) {
-            const newLanguages = this.state.suggestedLanguages || [];
-            newLanguages.push(language);
-            this.setState({suggestedLanguages: newLanguages});
-          }
-        } catch (error) {
-          console.error(`Language ${languageCode} not found 2`);
-        }
-      });
+      try {
+        const languages = await Promise.all(
+          this.props.suggestedLanguages.map(async (languageCode) => {
+            try {
+              return await languagesStore.getLanguage(languageCode);
+            } catch (error) {
+              console.error(`Language ${languageCode} not found 2`);
```

Comment:
typo?
```suggestion
              console.error(`Language ${languageCode} not found`);
```

Reply:
If you search for `} not found ` in the codebase, you will see we have 6 identical errors. John added the numbers so it's clear which one is which when looking into them in Sentry

Reply:
Oh ok, we'd rather write something descriptive instead (like the method name calling it) but that's out of scope of this PR.

Reply:
Agree. I made a task for this: https://linear.app/kobotoolbox/issue/DEV-18/rename-numbered-language-x-not-found-1-errors-to-be-more-descriptive